### DISCUSSION TITLE
ICP odometry not giving good results in the case of rotatory movements

### DISCUSSION THREAD

**Discussion by zafaryab-a** [introlab/rtabmap#1050] [General]
In case of Translational  motion the result of ICP odometry in Quit good, but in case of rotational motion the result is very poor.
On giving the wheel odometry as guess, In case of rotational movement if wheel doesn't slips the ICP odom result is ok but if wheel slips then ICP odometry is not able to correct it.
Is there any way to improve the result of ICP odometry in rotational motion?

Following are the params i am using:

param name="scan_normal_k" type="int" value="0"
param name="scan_normal_radius" type="double" value="2.0"
param name="Icp/MaxTranslation" type="int" value="10"
param name="Icp/MaxRotation" type="double" value="3.0"
param name="Icp/CorrespondenceRatio" type="double" value="0.05"
param name="Icp/Iterations" type="int" value="5"
param name="scan_voxel_size" type="double" value="0.1"

I have reduced Icp/iterations to reduce the computation.

**Comment by matlabbe** [comment]
Which kind of lidar are you using? 2D or 3D?

Note that for 3D lidars, deskewing option of `icp_odometry` is not enabled by default. For both 2D and 3D lidars, deskewing will work best if you use an IMU. You may combine your wheel odometry with imu with [robot_localization](http://docs.ros.org/en/noetic/api/robot_localization/html/index.html), then the resulting odom fed as guess to `icp_odometry` to have more accurate orientation estimation, thus better deskewing (then better icp odometry estimation on rotation).
